src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3381,25 +3374,23 @@ private BoundExpression BindNullCoalescingOperator(BinaryExpressionSyntax node,
                 optLeftType.GetNullableUnderlyingType() :
                 optLeftType;
 
-            // SPEC ERROR: The spec does not call out that the left-hand side may not be a method group
-            // SPEC ERROR: or anonymous function. We should add a line to the spec to say so.
-
+            // SPEC: The left hand side must be either the null literal, or not a non-nullable value type. Method groups
```

Comment:
`..., or a nullable value type.`

Reply:
What about type parameters?

---
In reply to: [213723498](https://github.com/dotnet/roslyn/pull/29577#discussion_r213723498) [](ancestors = 213723498)

Reply:
Then perhaps `"..., or a nullable value type, or an unconstrained type parameter."`

---
In reply to: [213730337](https://github.com/dotnet/roslyn/pull/29577#discussion_r213730337) [](ancestors = 213730337,213723498)

Reply:
Both of these things are `not a non-nullable value type`

---
In reply to: [213757765](https://github.com/dotnet/roslyn/pull/29577#discussion_r213757765) [](ancestors = 213757765,213730337,213723498)

Reply:
@cston @jaredpar if you think the original wording is hard to read with the double negative, let me know. Otherwise, the existing wording covers these cases.

---
In reply to: [213770268](https://github.com/dotnet/roslyn/pull/29577#discussion_r213770268) [](ancestors = 213770268,213757765,213730337,213723498)